src/error.rs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Error types for datasketches operations
+
+use std::fmt;
+
+/// Result type for datasketches serialization / deserialization operations
+pub type SerdeResult<T> = std::result::Result<T, SerdeError>;
```

Comment:
What if we remove this alias and use `Result<T, SerdeError>` everywhere?

As far as I can say, downstream projects would seldom use `SerdeResult` - they care about the Error and highly possibly convert to their own error type, making SerdeResult less useful.

Take `std::io::Error` as an example; there is no `std::io::Result`. And ditto `serde::se/der::Error`.